nimble/phy: Add initial support for nRF54L15 PHY

mkasenberg opened the pull request.

andrzej-kaczmarek commented:
seems like lost of the code was copied from the nrf5x driver so this should be part of that driver, i.e. the same way as we have nrf52 and nrf53

mkasenberg committed:
nimble/phy: Add initial support for nRF54L15 PHY

m-gorecki committed:
nimble/phy: Add an initial encryption for nRF54L15

sjanc commented on nimble/drivers/nrf5x/src/nrf53/phy_hw.h:
```diff
@@ -0,0 +1,20 @@
+ * under the License.
+ */
+
+#include "../nrf52/phy_hw.h"
```
I don't like this approach, although I don't have better idea for the moment (other than common phy_hw with #ifdefs...)   :)

sjanc reviewed (commented):
package should be name `nrf54l` since L15,L10 and L5 share same radio

mkasenberg commented:
I force-pushed the version without the static inline functions because I had @m-gorecki patch applied to it. Should I keep this version or switch to static inline? I remember there was a discussion about the readability of this version, but I don't remember what was the conclusion.

I use this version for Channel Sounding testing.

mkasenberg committed:
nimble/phy: nRF54L15: Add workaround to use RTC10 on nrfx 3.14.0

The RTC for nRF54L15 has been removed since nrfx 3.10.0, but at least
older board revisions (like 1.0.0) can still handle RTC at the old
address.

mkasenberg commented:
UPF tested
